include/zephyr/bluetooth/bluetooth.h

```diff
@@ -611,11 +616,12 @@ struct bt_le_local_features {
 	/**
 	 * @brief Local LE controller supported features.
 	 *
+	 * Pointer to array of size BT_LE_LOCAL_SUPPORTED_FEATURES_SIZE.
 	 * Refer to BT_LE_FEAT_BIT_* for values.
-	 * Refer to the BT_FEAT_LE_* macros for value comparionson.
+	 * Refer to the BT_FEAT_LE_* macros for value comparison.
 	 * See Bluetooth Core Specification, Vol 6, Part B, Section 4.6.
 	 */
-	uint8_t features[BT_LE_LOCAL_SUPPORTED_FEATURES_SIZE];
+	const uint8_t *features;
```

Comment:
I think this is a breaking API change. Please keep it as an array for now, and if there a significant optimization to be done here, we can open a separate PR to discuss this.

Otherwise, we need to add a point to the migration guide and document a lifetime for this pointer.